src/hooks.rs

```diff
@@ -0,0 +1,59 @@
+use std::{process::Stdio, sync::Arc};
+
+use anyhow::{bail, Context, Result};
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+use serde::Deserialize;
+use tokio::process::Command;
+
+use crate::config::RtcBuild;
+
+pub async fn spawn_hooks(cfg: Arc<RtcBuild>, stage: HookStage) -> Result<()> {
```

Comment:
Maybe just a doc string `///` describing what this does, especially the behavior related to only spawning hooks at the given stage.

Reply:
Done!